src/react-components/button-link.tsx

```diff
@@ -14,13 +14,15 @@ export interface ButtonLinkProps
   href?: string | undefined
   icon?: string
   className?: string
+  position?: "left" | "right" | "center"
```

Comment:
I'm a bit nitpicky on this since position doesn't conform here to a flexbox css understanding. 

But I think for this case we can accept it like this. Below I have some alternative solution which we can use in the future for more complex behaviours like aligning vertically and horizontally. 

---

I would use a new buttonStyle here for this div positioning and have variants which can create a type for you so you can extend the interface like so https://github.com/ory/elements/blob/main/src/react-components/button.tsx#L4-L13

You can then call the variants key `justifyContent` with the options `left` | `right` | `center`

e.g.

```typescript
export const buttonPositionStyle = recipe({
  base: {}
  variants: {
     justifyContent: {
       left: { 
         justifyContent: "left" 
       }
     }
   }
})

// Get the type
export type ButtonPositionStyle = RecipeVariants<typeof buttonPositionStyle>
```

Then extend the new type inside the button interface

```typescript
// required since interfaces cannot extend types whose properties are not statically known
type buttonPositionStyle = ButtonPositionStyle & Record<string, unknown>

export interface ButtonLinkProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement>.
  buttonPositionStyle {
  children?: React.ReactNode
  href?: string | undefined
  icon?: string
  className?: string
}
```

Also what about things like line-height when the button is next to some text? Shouldn't it always contain the `align-items: center;`?

Reply:
Done.

Reply:
>// required since interfaces cannot extend types whose properties are not statically known

BTw. any reason why you're using interfaces and not just union types?

Reply:
Not sure if you can use unions with an interface since we always extend the `React.HTMLAttributes` interface for additional "standard" html properties.

Reply:
You can, I did it here to test it out, as I prefer types to interfaces (but that's just personal preference): https://github.com/ory/elements/pull/41/files#diff-b968003ffef0642bdf3e2481d43954d2d65f467fbd905818b4921b7f3da4d84fR21